app/utils.py

```diff
@@ -17,49 +18,39 @@ def __init__(self, bot_path, item_type):
             )
         )
         with open(config_file, "r") as file:
-            self.configs = yaml.safe_load(file)
+            configs = yaml.safe_load(file)
 
-        if self.configs is None:
+        if configs is None:
             raise ValueError("No config found in the file")
 
-        self.list = self.configs.get(item_type)
+        self.list = configs.get(self.item_type)
         if self.list is None:
-            raise ValueError(f"No {item_type} found in the config file")
-
-        for list_items in self.list:
-            if list_items.get("media") is None:
-                raise ValueError(
-                    f"No media found in the config file for {list_items} in {item_type}"
-                )
-            elif list_items.get("format") is None:
-                raise ValueError(
-                    f"No format found in the config file for {list_items} in {item_type}"
-                )
+            raise ValueError(f"No {self.item_type} found in the config file")
+
+        errors = [
+            f"No media found in the config file for {item} in {self.item_type}"
+            for item in self.list
+            if item.get("media") is None
+        ] + [
+            f"No format found in the config file for {item} in {self.item_type}"
+            for item in self.list
+            if item.get("format") is None
+        ]
+        if errors:
+            raise ValueError("\n".join(errors))
 
         access_token = os.environ.get("GALAXY_SOCIAL_BOT_TOKEN")
         repo_name = os.environ.get("REPO")
         g = Github(access_token)
         self.repo = g.get_repo(repo_name)
 
-        self.existing_files = set(
-            pr_file.filename
-            for pr in self.repo.get_pulls(state="open")
-            for pr_file in pr.get_files()
-            if pr_file.filename.startswith(self.bot_path)
-        )
-        git_tree = self.repo.get_git_tree(self.repo.default_branch, recursive=True)
-        self.existing_files.update(
-            file.path
-            for file in git_tree.tree
-            if file.path.startswith(self.bot_path) and file.path.endswith(".md")
-        )
-        self.branch_name = (
-            f"{self.bot_path}-update-{datetime.now().strftime('%Y%m%d%H%M%S')}"
-        )
-        self.repo.create_git_ref(
-            ref=f"refs/heads/{self.branch_name}",
-            sha=self.repo.get_branch("main").commit.sha,
-        )
+        self.existing_files = {
```

Comment:
I don't find this very readable. Why not pull the for-loop out and construct a set inside the for loop?